### PR TITLE
[paddle] concat

### LINKED ISSUE #16543: concat

### PR DESCRIPTION
**Pull request by huzi6181**
#16543

please review

### CODE REVIEW

**Comment by huzi6181** [comment]
This is failing tests on packages that i didnt even work on, please advise.

**Review by khushi-411 — COMMENTED** [review]
Hi @huzi6181,
Thanks for the PR! Yes, many failures are not related to the PR. That's due to the current CI status. Could you please resolve the conflicts and update the PR? Thanks!
Please feel free to reach out, if you need any heads up!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 1]
```diff
@@ -125,12 +124,12 @@ def test_paddle_tensor_property_ndim(
...
-    init_flags,
-    method_flags,
-    frontend,
-    on_device,
+        dtype_x_shape,
+        frontend_method_data,
+        init_flags,
+        method_flags,
+        frontend,
+        on_device,
```
Comment (khushi-411) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:132]: These formatting are not related to the scope of the PR. Please undo the changes.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 2]
```diff
@@ -163,12 +162,12 @@ def test_paddle_instance_reshape(
...
-    init_flags,
-    method_flags,
-    frontend,
-    on_device,
+        dtype_x_index,
+        frontend_method_data,
+        init_flags,
+        method_flags,
+        frontend,
+        on_device,
```
Comment (khushi-411) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:170]: ditto

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by huzi6181** [comment]
Im working on it

**Comment by huzi6181** [comment]
@khushi-411 ive resolved the formatting issues, can you please check now

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ def sinh(self, name=None):
     @with_unsupported_dtypes({"2.4.2 and below": ("float16", "bfloat16")}, "paddle")
     def argmax(self, axis=None, keepdim=False, dtype=None, name=None):
         return ivy.argmax(self._ivy_array, axis=axis, keepdims=keepdim, dtype=dtype)
-
+
```
Comment (khushi-411) [inline ivy/functional/frontends/paddle/tensor/tensor.py:128]: nit: we don't need to add spaces here

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 4]
```diff
@@ -155,3 +155,9 @@ def floor(self, name=None):
     )
     def tanh(self, name=None):
         return ivy.tanh(self._ivy_array)
+    
+    @with_unsupported_dtypes(
+        {"2.4.2 and below": ("float16", "bfloat16")}, "paddle"
+    )
+    def concat(self, x, axis=0, name=None):
+        return ivy.concat(x, axis)
```
Comment (khushi-411) [inline ivy/functional/frontends/paddle/tensor/tensor.py:163]: Why did you add `self` and `x` both as an argument? We need one input tensor, which needs to be concatenated.

The best way is to check the docs if you need clarification. Here's the link: https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/fluid/layers/tensor/concat_en.html

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 5]
```diff
@@ -7,7 +7,6 @@
 from ivy_tests.test_ivy.helpers import handle_frontend_method
 from ivy.functional.frontends.paddle import Tensor
 
-
```
Comment (khushi-411) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:10]: nit: please undo these changes; they are not related to the PR scope, nor are they required.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 6]
```diff
@@ -66,7 +65,7 @@ def _setitem_helper(draw, available_dtypes, allow_neg_step=True):
     ).filter(lambda x: "bfloat16" not in x[0]),
 )
 def test_paddle_tensor_property_device(
-    dtype_x,
+        dtype_x,
```
Comment (khushi-411) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:68]: nit: we don't need to add extra spaces here; please undo the changes

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 7]
```diff
@@ -82,7 +81,7 @@ def test_paddle_tensor_property_device(
     ).filter(lambda x: "bfloat16" not in x[0]),
 )
 def test_paddle_tensor_property_dtype(
-    dtype_x,
+        dtype_x,
 ):
```
Comment (khushi-411) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:85]: nit: same as above

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 8]
```diff
@@ -110,7 +109,7 @@ def test_paddle_tensor_property_shape(dtype_x):
     ).filter(lambda x: "bfloat16" not in x[0]),
 )
 def test_paddle_tensor_property_ndim(
-    dtype_x,
+        dtype_x,
 ):
```
Comment (khushi-411) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:113]: nit: same as above

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 9]
```diff
@@ -128,7 +127,7 @@ def test_paddle_instance_reshape(
     dtype_x_shape,
     frontend_method_data,
     init_flags,
-    method_flags,
+    method_flags
```
Comment (khushi-411) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:130]: nit: we still need a comma here

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 10]
```diff
@@ -727,3 +726,42 @@ def test_paddle_tanh(
...
+    init_flags,
+    method_flags,
+    frontend,
+    on_device,
+):
+    input_dtype, x, axis = dtype_x_axis
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtype,
+        init_flags=init_flags,
+        init_all_as_kwargs_np={
```
Comment (khushi-411) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:755]: Please correct the tests according to the functions you'll modify.

There are some great docs on [testing](https://unify.ai/docs/ivy/docs/ivy_tests.test_ivy.helpers.html) and [YouTube](https://www.youtube.com/watch?v=iS7QFsQa9bI) videos, which will clear your concepts. Just thought to share if you need any heads-up.

**Review by khushi-411 — CHANGES_REQUESTED** [review]
Hi, @huzi6181! Thanks for the PR. There are some minor nitpicks you need to work on, as well as you need to correct the function definition. You defined with four args; conversely, it takes one input arg, the axis arg, and an optional name arg. Overall it looks like a great start; please ping me if you need help!

Thanks!

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 4 above)
Comment (huzi6181) [reply ivy/functional/frontends/paddle/tensor/tensor.py:163]: I defined the function as a method in the tensor class which why i added self.
